ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,624 @@
+                    ?association    OBO:RO_has_environment ?environment   .
+                    OPTIONAL { ?association  dc:source ?source } .
+                    ?association    OBAN:association_has_subject ?feature .
+                    ?association    OBAN:association_has_object ?phenotype .
+                    ?environment  rdfs:label ?environment_label  .
+                    ?phenotype  rdfs:label ?phenotype_label  .
+                    ?feature  rdfs:label ?feature_label  .
+                    OPTIONAL { ?feature owl:sameAs  ?external_id } .
+                    OPTIONAL { ?phenotype BFO:0000159 ?phenotype_quality } .
+                    #%FILTER%
+                    }
+            GROUP  BY ?association
+            ORDER  BY ?association
+"""
+
+    def _toNamespaceURL(self, term):
+        """
+        Given an ontologyterm.term return namespace identifier.
+        Leverages prefixes already in graph namespace
+        Ex.  "DrugBank:DB01268" -> "http://www.drugbank.ca/drugs/DB01268"
+        """
+        termPrefix, termId = term.split(':')
+        for prefix, namespace in self._rdfGraph.namespaces():
+            if prefix == termPrefix:
+                return "".join([namespace, termId])
+        raise exceptions.NotImplementedException(
+           "Term has a prefix not found in this instance. {}".format(term))
+
+    def _getIdentifier(self, url):
+        """
+        Given a url identifier return identifier portion
+        Leverages prefixes already in graph namespace
+        Returns None if no match
+        Ex.  "http://www.drugbank.ca/drugs/DB01268" -> "DB01268"
+        """
+        for prefix, namespace in self._rdfGraph.namespaces():
+            if namespace in url:
+                return url.replace(namespace, '')
+
+    def _getPrefix(self, url):
+        """
+        Given a url return namespace prefix.
+        Leverages prefixes already in graph namespace
+        Ex.  "http://www.drugbank.ca/drugs/" -> "Drugbank"
+        """
+        for prefix, namespace in self._rdfGraph.namespaces():
+            if namespace.toPython() == url or namespace == url:
+                return prefix
+        raise exceptions.NotImplementedException(
+           "No namespace found for url {}".format(url))
+
+    def _getPrefixURL(self, url):
+        """
+        Given a url return namespace prefix.
+        Leverages prefixes already in graph namespace
+        Ex.  "http://www.drugbank.ca/drugs/DDD"
+            -> "http://www.drugbank.ca/drugs/"
+        """
+        for prefix, namespace in self._rdfGraph.namespaces():
+            if namespace.toPython() in url:
+                return namespace
+
+    def _toGA4GH(self, association):
+        """
+        given an association dict,
+        return a protocol.FeaturePhenotypeAssociation
+        """
+        # TODO: This method needs to broken in several parts. It's
+        # doing too much things
+        fpa = None
```

Comment:
What's this fpa = None for? In general this method looks a bit messy. I think if we had a quick  inline comment describing what each section is for, and remove any dead code and comments, then it would be a big improvement.